coreos-installer

```diff
@@ -764,52 +765,16 @@ EOF
     done
 }
 
-#########################################################
-#Handle s390x zFCP SCSI devices
-#########################################################
-write_image_to_zfcp_disk() {
-    log "Extracting disk image"
-    mount_tmpfs $(( $($DECOMPRESSION_TOOL ${imagefile_compressed} | wc --bytes) / 1024 / 1024 + ${TMPFS_MBSIZE} + 100 ))
-    $DECOMPRESSION_TOOL ${imagefile_compressed} > ${imagefile}
-
-    # we have to store the strings to a file because any change in this
-    # while-do block would not take effect outside
-    (while IFS='' read -r line || [ -n "$line" ]
-    do
-        if [[ $line =~ "start=" ]]
-        then
-            # we save the start= and size= here, then dd the partitions later
-            printf "%s:%s\n" \
-                "$(echo $line | cut -d: -f 2 | cut -d, -f 1 | tr -d ' ')" \
-                "$(echo $line | cut -d: -f 2 | cut -d, -f 2 | tr -d ' ')" \
-                >> /tmp/sfdisk_info
-        elif ! [[ $line =~ "label:" || $line =~ "label-id:" || $line =~ "unit:" ]]
-        then
-            continue
-        fi
-        echo $line
-    done <<< $(sfdisk -d ${imagefile})
-    ) | sfdisk "${DEST_DEV}"
-
-    while IFS='' read -r line || [ -n "$line" ]
-    do
-        eval ${line%%:*} ${line##*:}
-        dd if=${imagefile} bs=512 iflag=fullblock \
-            of="${DEST_DEV}" status=none skip=$start seek=$start count=$size
-        if [[ $? -ne 0 ]]; then
-            log "failed to write image to zFCP SCSI disk"
-            exit 1
-        fi
-    done < /tmp/sfdisk_info
-}
 #########################################################
 #And Write the image to disk
 #########################################################
 write_image_to_disk() {
     log "Writing disk image"
 
     set -o pipefail
-    if [[ "${arch}" =~ (aarch64|ppc64le|x86_64) ]]
+    # only s390x DASD devices require a special code path
+    if [[ "${arch}" =~ (aarch64|ppc64le|x86_64) || \
+         ! ( "${arch}" = s390x && "${DEST_DEV}" =~ ^.*dasd.*$ ) ]]
```

Comment:
Hmm, is this conditional correct? Seems like it'll be true for *any* basearch which isn't s390x, not just the specific whitelisted ones. I think we want something like:

```
    if [[ "${arch}" =~ (aarch64|ppc64le|x86_64) || \
        ( "${arch}" == s390x && [[ ! "${DEST_DEV}" =~ ^.*dasd.*$ ]] ) ]]
```

right?